cvmfs/network/download.cc

```diff
@@ -1048,12 +982,21 @@ void DownloadManager::SetUrlOptions(JobInfo *info) {
     url = ReplaceAll(url, "@proxy@", replacement);
   }
 
-  if ((info->destination == kDestinationMem) &&
-      (info->destination_mem.size == 0) &&
-      HasPrefix(url, "file://", false))
-  {
-    info->destination_mem.size = 64*1024;
-    info->destination_mem.data = static_cast<char *>(smalloc(64*1024));
+  // TODO(heretherebedragons) how do we make size_ == 0 generic?
+  // MAYBE have an empty non-functional sink, so that you do not have to check
+  // for kDestinationNone???
+  if ((info->destination != kDestinationNone) &&
+      info->destination_sink->RequiresReserve() &&
+      (static_cast<cvmfs::MemSink*>(info->destination_sink)->size_ == 0) &&
+      HasPrefix(url, "file://", false)) {
+    platform_stat64 stat_buf;
+    int retval = platform_stat(url.c_str(), &stat_buf);
+    if (retval != 0) {
+      // TODO(heretherebedragons) WHAT DO WE DO IN AN ERROR CASE?
+      // info->destination_sink->Reserve(64ul * 1024ul);
```

Comment:
Let's generally reserve 64kB. You already have the code to increase the buffer as needed.

Reply:
so do  you want to remove the part again to allocate as much space as needed for a local file?

https://github.com/cvmfs/cvmfs/blob/943eff2c581788030cb3f679d3e7cc64cc3d8a85/cvmfs/network/download.cc#LL998C1-L998C1